### PR TITLE
chore: add better regclass and oid cast support

### PR DESCRIPTION
**Pull request by universalmind303**
partially addresses https://github.com/GlareDB/glaredb/issues/2746

### CODE REVIEW

File: testdata/sqllogictests/cast/oid.slt [hunk 1]
```diff
@@ -0,0 +1,30 @@
+
+statement error
```
Comment (tychoish) [inline testdata/sqllogictests/cast/oid.slt:2]: should we put the error message here?

File: testdata/sqllogictests/cast/oid.slt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+statement ok
+select cast('my_table' as OID);
+
+
+# We don't support expressions other than strings right now.
+statement error Casting expressions to oid unsupported
+select cast(1 as oid);
+
+statement error Casting expressions to oid unsupported
```
Comment (tychoish) [inline testdata/sqllogictests/cast/oid.slt:26]: should this error message be phrased as "only strings can be cast to OIDs"

File: testdata/sqllogictests/cast/oid.slt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+select 'my_table'::OID;
+
+statement ok
+select cast('my_table' as oid);
+
+statement ok
+select cast('my_table' as OID);
+
+
+# We don't support expressions other than strings right now.
```
Comment (tychoish) [inline testdata/sqllogictests/cast/oid.slt:22]: is there a point to supporting more ever/eventually?

**Review by tychoish — COMMENTED** [review]
few questions, seems reasonable. 

(shouldn't this be a fix not a chore?)

File: testdata/sqllogictests/cast/oid.slt (hunk 3 above)
Comment (universalmind303) [reply testdata/sqllogictests/cast/oid.slt:22]: idk, most of this was copied from the `regclass` stuff, so i didn't want to change any more than absolutely necessary to implement the new functionality

**Comment by tychoish** [comment]
The existing regclass error is only used once, and is ambiguous from the user's perspective (and ours, when debugging) if they ever run into it, particularly now that we (presumably) support integer and string casts. If it's easy enough to provide a bit more context, it seems worthwhile. 

I don't see any particular reason to treat our error messages as a stable API, unless I'm missing something or this is a standard/copied error message.

**Review by vrongmeal — APPROVED** [review]

**Pull request merged**